### PR TITLE
[Synapse] Add cmdlet to encrypt credential in linked service 

### PR DESCRIPTION
**Pull request by v-hozhao**
## Description

Encrypt credential in linked service with specified integration runtime.

- New-AzSynapseLinkedServiceEncryptedCredential

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - [[Synapse] Add cmdlet to encrypt credential in linked service #1015](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1015)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Synapse/Synapse/Az.Synapse.psd1 [hunk 1]
```diff
@@ -175,12 +175,13 @@ CmdletsToExport = 'Get-AzSynapseSparkJob', 'Stop-AzSynapseSparkJob',
...
-               'New-AzSynapseManagedPrivateEndpoint', 
-               'Remove-AzSynapseManagedPrivateEndpoint', 
+               'Set-AzSynapseSparkJobDefinition',
+               'Get-AzSynapseManagedPrivateEndpoint',
+               'New-AzSynapseManagedPrivateEndpoint',
+               'Remove-AzSynapseManagedPrivateEndpoint',
                'New-AzSynapseGitRepositoryConfig', 'New-AzSynapseWorkspacePackage', 
-               'Get-AzSynapseWorkspacePackage', 'Remove-AzSynapseWorkspacePackage'
+               'Get-AzSynapseWorkspacePackage', 'Remove-AzSynapseWorkspacePackage',
+			         'New-AzSynapseLinkedServiceEncryptedCredential'
```
Comment (idear1203) [inline src/Synapse/Synapse/Az.Synapse.psd1:184]: >			 [](http://example.com/codeflow?start=0&length=3)

change tabs to spaces.

**Comment by idear1203** [comment]
> 

Add 'Microsoft.DataTransfer.Gateway.Encryption.dll' below

---
Refers to: src/Synapse/Synapse/Az.Synapse.psd1:64 in ecccc14. [](commit_id = ecccc142ad024952341b651d310f9b6d328df8e6, deletion_comment = False)

File: src/Synapse/Synapse/ChangeLog.md [hunk 2]
```diff
@@ -36,6 +36,7 @@
 	- Added `Remove-AzSynapseWorkspacePackage` cmdlet
 	- Updated `New-AzSynapseSparkPool` cmdlet to drop parameter `-LibraryRequirementsFilePath`
 	- Updated `Updated-AzSynapseSparkPool` cmdlet to add parameter `-Package` and `-PackageAction`
+* Added `New-AzSynapseLinkedServiceEncryptedCredential` cmdlet to encrypt credential in linked service
```
Comment (idear1203) [inline src/Synapse/Synapse/ChangeLog.md:40]: This line should go to Upcoming Release section

File: src/Synapse/Synapse/Synapse.csproj [hunk 3]
```diff
@@ -11,11 +11,12 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Azure.Analytics.Synapse.AccessControl" Version="1.0.0-preview.3" />
-    <PackageReference Include="Azure.Analytics.Synapse.Artifacts" Version="1.0.0-preview.10" />
     <PackageReference Include="Azure.Analytics.Synapse.ManagedPrivateEndpoints" Version="1.0.0-beta.3" />
-    <PackageReference Include="Azure.Analytics.Synapse.Spark" Version="1.0.0-beta.1" />
     <PackageReference Include="Microsoft.Azure.Management.Synapse" Version="2.1.0-preview" />
+    <PackageReference Include="Azure.Analytics.Synapse.Spark" Version="1.0.0-beta.1" />
+    <PackageReference Include="Azure.Analytics.Synapse.AccessControl" Version="1.0.0-preview.3" />
+    <PackageReference Include="Azure.Analytics.Synapse.Artifacts" Version="1.0.0-preview.9" />
```
Comment (idear1203) [inline src/Synapse/Synapse/Synapse.csproj:18]: >1.0.0-preview.9 [](http://example.com/codeflow?start=75&length=15)

it should be 1.0.0-preview.10.

Comment (idear1203) [inline src/Synapse/Synapse/Synapse.csproj:18]: Please keep these lines to be exactly same with that of main branch.

File: src/Synapse/Synapse/ChangeLog.md (hunk 2 above)
Comment (v-hozhao) [reply src/Synapse/Synapse/ChangeLog.md]: done

File: src/Synapse/Synapse/Az.Synapse.psd1 (hunk 1 above)
Comment (v-hozhao) [reply src/Synapse/Synapse/Az.Synapse.psd1]: done

File: src/Synapse/Synapse/Synapse.csproj (hunk 3 above)
Comment (v-hozhao) [reply src/Synapse/Synapse/Synapse.csproj]: Done

File: src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/LinkedServices/NewAzSynapseLinkedServiceEncryptedCredential.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+using Microsoft.Azure.Commands.Synapse.Common;
+using Microsoft.Azure.Commands.Synapse.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Globalization;
+using Microsoft.Azure.Commands.Synapse.Properties;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Synapse
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + SynapseConstants.SynapsePrefix + SynapseConstants.LinkedServiceEncryptedCredential,
```
Comment (wyunchi-ms) [inline src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/LinkedServices/NewAzSynapseLinkedServiceEncryptedCredential.cs:29]: Need to support `ShouldProcess` for `New` cmdlet.

Comment (v-hozhao) [reply src/Synapse/Synapse/Commands/DataPlaneCommands/Artifact/LinkedServices/NewAzSynapseLinkedServiceEncryptedCredential.cs:29]: Done

File: src/Synapse/Synapse/Synapse.csproj (hunk 3 above)
Comment (v-hozhao) [reply src/Synapse/Synapse/Synapse.csproj]: Done

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**